packages/expo-updates/src/UpdatesProvider.utils.ts

```diff
@@ -0,0 +1,103 @@
+import * as Updates from './Updates';
+import type { Manifest, UpdateEvent } from './Updates.types';
+import { UpdatesProviderEventType, currentlyRunning } from './UpdatesProvider.constants';
+import type {
+  AvailableUpdateInfo,
+  UpdatesInfo,
+  UpdatesProviderEvent,
+} from './UpdatesProvider.types';
+
+/////// Internal functions ////////
+
+// Constructs the availableUpdate from the update manifest
+export const availableUpdateFromManifest = (manifest: Partial<Manifest> | undefined) => {
+  return manifest
+    ? {
+        updateId: manifest?.id ? manifest?.id : null,
+        createdAt: manifest?.createdAt ? new Date(manifest?.createdAt) : null,
+        manifest,
+      }
+    : undefined;
+};
+
+// Constructs the UpdatesInfo from an event
+export const updatesInfoFromEvent = (event: UpdateEvent): UpdatesInfo => {
+  const lastCheckForUpdateTime = new Date();
+  if (event.type === Updates.UpdateEventType.NO_UPDATE_AVAILABLE) {
```

Comment:
nit: use a switch to typescript enforces exhaustiveness

Reply:
I will do that in a future PR. Since this PR will be cherry-picked for SDK 48, I'm only supporting the event types that exist there.